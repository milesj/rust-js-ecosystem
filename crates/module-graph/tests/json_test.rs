mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod json {
    use super::*;

    #[test]
    fn object() {
        let sandbox = create_sandbox("json");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "object.json"));
    }

    #[test]
    fn array() {
        let sandbox = create_sandbox("json");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "array.json"));
    }

    #[test]
    fn primitive() {
        let sandbox = create_sandbox("json");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "primitive.json"));
    }
}
