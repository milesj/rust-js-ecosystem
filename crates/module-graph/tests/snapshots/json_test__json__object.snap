---
source: crates/module-graph/tests/json_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"object.json\")"
---
{
    1: Module {
        exports: [
            Export {
                kind: Native,
                module_id: None,
                source: None,
                span: None,
                symbols: [
                    ExportedSymbol {
                        kind: Default,
                        symbol_id: None,
                        name: "default",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "a",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "b",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "c",
                    },
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "d",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/object.json",
        query: None,
        source: JsonModule {
            data: Object {
                "a": Number(123),
                "b": String("abc"),
                "c": Bool(true),
                "d": Object {
                    "e": Number(456),
                    "f": String("def"),
                    "g": Bool(false),
                },
            },
            source: "{\n  \"a\": 123,\n  \"b\": \"abc\",\n  \"c\": true,\n  \"d\": {\n    \"e\": 456,\n    \"f\": \"def\",\n    \"g\": false\n  }\n}\n",
        },
    },
}
